broker/lib/fabrik/DBManager.js

```diff
@@ -214,7 +231,21 @@ class DBManager {
           logger.info(`MongoDB ${operation} request is complete. Check status for task id - ${taskId}`);
           this.director
             .pollTaskStatusTillComplete(taskId)
-            .then(response => this.dbCreateUpdateSucceeded(response, createIfNotPresent))
+            .then(response => {
+              return Promise.try(() => {
+                  if (createIfNotPresent) {
+                    //This is precaution to ensure that no previous bind resource exists in the event that
+                    //service-fabrik-mongodb is recreated but ApiServer is not.
+                    return eventmesh.apiServerClient.deleteResource({
+                      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BIND,
+                      resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR_BIND,
+                      resourceId: _.toLower(CONST.FABRIK_INTERNAL_MONGO_DB.BINDING_ID)
+                    });
+                  }
+                })
+                .then(() => this.dbCreateUpdateSucceeded(response, createIfNotPresent))
+                .catch(NotFound, () => this.dbCreateUpdateSucceeded(response, createIfNotPresent));
```

Comment:
can move this `catch(NotFound), ()=> logger.debug('DB binding not present. ignoring it'))` to LN243 to avoid dupe `dbCreateUpdateSucceeded`

Reply:
This will be resolved if this logic moves to storeBindingProperty method.